packages/astro/src/runtime/client/media.ts

```diff
@@ -5,7 +5,23 @@ import type { GetHydrateCallback, HydrateOptions } from '../../@types/astro';
  */
 export default async function onMedia(astroId: string, options: HydrateOptions, getHydrateCallback: GetHydrateCallback) {
 	const roots = document.querySelectorAll(`astro-root[uid="${astroId}"]`);
-	const innerHTML = roots[0].querySelector(`astro-fragment`)?.innerHTML ?? null;
+	if(roots.length === 0) {
+		throw new Error(`Unable to find the root for the component ${options.name}`);
+	}
+
+	let innerHTML: string | null = null;
+  let fragment = roots[0].querySelector(`astro-fragment`);
+  if(fragment == null) {
+		// If there is no child fragment, check to see if there is a template.
+		// This happens if children were passed but the client component did not render any.
+    let template = roots[0].querySelector(`template[data-astro-template]`);
```

Comment:
Note to self, this is going to check for this template any time there is not an astro-fragment. We can probably short circuit this check so it's only done when needed.